### PR TITLE
Code Quality: Added Save dialog project

### PR DESCRIPTION
**Pull request by gave92**
TODO:
- [x] Enable only for debug mode

### CODE REVIEW

File: src/Files.App/Views/Settings/AdvancedPage.xaml [hunk 1]
```diff
@@ -162,8 +162,7 @@
 			<local:SettingsBlockControl
 				x:Name="ReplaceOpenFileDialogSetting"
 				Title="{helpers:ResourceString Name=SettingsSetAsOpenDialog}"
-				HorizontalAlignment="Stretch"
-				x:Load="False">
+				HorizontalAlignment="Stretch">
```
Comment (yaira2) [inline src/Files.App/Views/Settings/AdvancedPage.xaml:165]:
```suggestion
				x:Load="False"
				HorizontalAlignment="Stretch">
```

Comment (yaira2) [reply src/Files.App/Views/Settings/AdvancedPage.xaml:165]: Alternatively we can keep it visible in debug mode

**Review by yaira2 — DISMISSED** [review]

File: src/Files.App/Views/Settings/AdvancedPage.xaml (hunk 1 above)
Comment (gave92) [reply src/Files.App/Views/Settings/AdvancedPage.xaml]: Done!

**Comment by yaira2** [comment]
LGTM

File: src/Files.App.OpenDialog/FilesOpenDialog.cpp [hunk 2]
```diff
@@ -23,7 +23,7 @@ CComPtr<IFileOpenDialog> GetSystemDialog()
 {
 	HINSTANCE lib = CoLoadLibrary(L"C:\\Windows\\System32\\comdlg32.dll", false);
 
-	BOOL(WINAPI *dllGetClassObject)(REFCLSID, REFIID, LPVOID*) = (BOOL(WINAPI*)(REFCLSID, REFIID, LPVOID*))GetProcAddress(lib, "DllGetClassObject");
+	BOOL(WINAPI * dllGetClassObject)(REFCLSID, REFIID, LPVOID*) = (BOOL(WINAPI*)(REFCLSID, REFIID, LPVOID*))GetProcAddress(lib, "DllGetClassObject");
```
Comment (0x5bfa) [inline src/Files.App.OpenDialog/FilesOpenDialog.cpp:26]:
```suggestion
	BOOL(WINAPI* dllGetClassObject)(REFCLSID, REFIID, LPVOID*) = (BOOL(WINAPI*)(REFCLSID, REFIID, LPVOID*))GetProcAddress(lib, "DllGetClassObject");
```

File: src/Files.App/App.xaml.cs [hunk 3]
```diff
@@ -11,6 +11,7 @@
 using Windows.ApplicationModel.DataTransfer;
 using Windows.Storage;
 using Windows.UI.Notifications;
+using static Files.App.Helpers.Win32PInvoke;
```
Comment (0x5bfa) [inline src/Files.App/App.xaml.cs:14]: I don't like using static. If we focus on merge, you can ignore this.

File: src/Files.App/Program.cs [hunk 4]
```diff
@@ -62,11 +62,17 @@ private static void Main()
 			var OpenTabInExistingInstance = ApplicationData.Current.LocalSettings.Values.Get("OpenTabInExistingInstance", true);
 			var activatedArgs = AppInstance.GetCurrent().GetActivatedEventArgs();
 
-			if (activatedArgs.Data is ICommandLineActivatedEventArgs cmdLineArgs)
+			// WINUI3: When launching from commandline the argument is not ICommandLineActivatedEventArgs (#10370)
```
Comment (0x5bfa) [inline src/Files.App/Program.cs:65]: I think this issue has been resolved.
Can you please confirm this code is still valid

**Review by 0x5bfa — CHANGES_REQUESTED** [review]
Looks ok overall.

File: src/Files.App/Program.cs (hunk 4 above)
Comment (gave92) [reply src/Files.App/Program.cs:65]: Yeah I've checked and this is still needed. Equivalent code is in MainWindow.cs

Comment (0x5bfa) [reply src/Files.App/Program.cs:65]: Thanks for the checking.

File: src/Files.App.OpenDialog/FilesOpenDialog.cpp (hunk 2 above)
Comment (gave92) [reply src/Files.App.OpenDialog/FilesOpenDialog.cpp:26]: It's automatic formatting from VS, it thinks it's a product operation 😅

Comment (gave92) [reply src/Files.App.OpenDialog/FilesOpenDialog.cpp]: *Fixed manually

File: src/Files.App/App.xaml.cs (hunk 3 above)
Comment (gave92) [reply src/Files.App/App.xaml.cs]: Removed the static using 👍

**Comment by yaira2** [comment]
@gave92 LGTM

That said, on 23H2 this appears to crash Notepad for both opening and saving.

**Comment by gave92** [comment]
Thanks! When you got time we can take a look to debug why it's not woking for you.

**Pull request merged**